src/backgrounddownloader.cpp

```diff
@@ -0,0 +1,152 @@
+#include "backgrounddownloader.h"
+
+#include <iostream>
+#include <QString>
+#include <QTimer>
+#include <QDebug>
+
+BackgroundDownloader::BackgroundDownloader(kiwix::Downloader* downloader)
+    :  mp_downloader(downloader)
+{
+    QThread* newThread = new QThread(); // no parent
+    m_thread.reset(newThread);
+    moveToThread(m_thread.get());
```

Comment:
This parentless thread and call to `moveToThread()` is key to ensuring that `BackgroundDownloader` slots are invoked on their own thread that is separate from the UI thread.